docs/modules/data-connections/pages/data-connections-configuration.adoc

```diff
@@ -74,21 +70,274 @@ config
+--
+====
+
+[[kafka]]
+=== Example Kafka Data Connection
+
+This example shows the configuration of a data connection to a single Kafka broker.
+
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+  <data-connection name="my-kafka-source">
+    <type>Kafka</type>
+    <properties>
+      <property name="bootstrap.servers">127.0.0.1:9092</property> <1>
+      <property name="key.deserializer">org.apache.kafka.common.serialization.IntegerDeserializer</property> <2>
+      <property name="key.serializer">org.apache.kafka.common.serialization.IntegerSerializer</property>
+      <property name="value.serializer">org.apache.kafka.common.serialization.StringSerializer</property>
+      <property name="value.deserializer">org.apache.kafka.common.serialization.StringDeserializer</property>
+      <property name="auto.offset.reset">earliest</property> <3>
+    </properties>
+    <shared>true</shared>
+  </data-connection>
+</hazelcast>
+----
+<1> (Required) Address of the Kafka consumer/producer  
+<2> (Optional) Automatic serializers/deserializers for keys and values in Kafka messages 
+<3> (Optional) Consumer behavior if the connection is interrupted
+--
+
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  data-connection:
+    my-kafka-source:
+      type: Kafka
+      properties:
+        bootstrap.servers: 127.0.0.1:9092 <1>
+        key.deserializer: org.apache.kafka.common.serialization.IntegerDeserialize <2> 
+        key.serializer: org.apache.kafka.common.serialization.IntegerSerializer
+        value.serializer: org.apache.kafka.common.serialization.StringSerializer
+        auto.offset.reset: earliest <3>
+      shared: true
+----
+<1> (Required) Address of the Kafka consumer/producer  
+<2> (Optional) Automatic serializers/deserializers for keys and values in Kafka messages 
+<3> (Optional) Consumer behavior if the connection is interrupted
+--
+
+Java::
++
+--
+[source,java]
+----
+config
+  .addDataConnectionConfig(
+    new DataConnectionConfig("my-kafka-source")
+      .setType("Kafka")
+      .setProperty("bootstrap.servers", "127.0.0.1:9092") <1>
+      .setProperty("key.deserializer", "org.apache.kafka.common.serialization.IntegerDeserialize") <2>
+      .setProperty("key.serializer", "org.apache.kafka.common.serialization.IntegerSerializer")
+      .setProperty("value.serializer", "org.apache.kafka.common.serialization.StringSerializer")
+      .setProperty("auto.offset.reset", "earliest") <3>
+      .setShared(true)
+  );
+----
+<1> (Required) Address of the Kafka consumer/producer  
+<2> (Optional) Automatic serializers/deserializers for keys and values in Kafka messages 
+<3> (Optional) Consumer behavior if the connection is interrupted
+--
+SQL::
++
+--
+Data connections created in SQL behave differently to those defined in members' configuration files or in Java.
+
+- You cannot persist SQL metadata on disk. This means that when a cluster restarts, data connections are not retained.
```

Comment:
This will be possible in EE 

See https://github.com/hazelcast/hazelcast-enterprise/pull/5945

Reply:
Updated as requested.